### PR TITLE
Aws ec2 exercise

### PR DESCRIPTION
**Pull request by mcvholloway**
I put together an exercise for the students to do (in groups) where they get to practice using ssh and several other command line tools to do (a very simple) task on an ec2 instance and to interact with s3 through the command line interface.
Please let me know your thoughts. I can set up an instance and credentials if you want to try it out yourself.

### CODE REVIEW

**Comment by MaryLvV** [comment]
This is a great addition!  

- It might be good to link to a resource like [this one](https://www.tutorialspoint.com/unix_commands/chmod.htm) (unless ssh is introduced prior to this exercise).
- Is it correct that this exercise comes after the AWS_03 slides? If so maybe linking to a shell command tutorial or cheat sheet like the one above in those slides makes more sense makes more sense.
- I know we've gone back and forth on `.yml` vs `.yaml` and there are arguments for both (.yaml is the official extension [supported by yaml.org](https://yaml.org/faq.html) while 3-letter extensions are a [throwback to a time](https://en.wikipedia.org/wiki/Filename_extension) when everything was constrained to 8.3 filename formats and is preferred by some. We all agree to use `.html` over `.htm` and I'd really love to get consistency at some point.

**Review by MaryLvV — APPROVED** [review]

**Comment by mcvholloway** [comment]
@MaryLvV 
Thanks for the chmod link - I added it to the directions.
My plan is to revise the AWS_03 slides. What I want to do is to have this part come before Sagemaker and to not give them management console access until after this exercise. I'm going to have this exercise come after _watching_ a demonstration, but not directly following along during the demo. I think that for this type of exercise, where the main goal is to demystify interacting via ssh, that they'll benefit much more from being with their group and having to experiment with the commands rather than just following along and copy/pasting from a demo.
As for the `.yml` vs. `.yaml`, I've been going off of the [Anaconda docs](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file).

File: AWS/ec2_practice/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+`ssh -i "mykeys.pem" -L 8888:localhost:8888 <DNS>`
+
+The DNS will start with ubuntu@ec2 and will be with the other information I shared with you.
+
+Breaking down what this does, we are first pointing ssh to our private key file with the -i flag. The -L flag sets forwards the 8888 port on our local machine to the 8888 port on the localhost of the ec2 instance. The purpose of this is to allow us to connect to a jupyter notebook running on the ec2 instance. Finally, we need to specify the DNS of the ec2 instance we are connecting to.
+
+The first time you do this, you will get a warning message that the authenticity of the host can't be established and will ask you to verify that you want to continue connecting.
+
+3. One you have successfully connected, we need to set up our instance in order to be able to use Python and the libraries we need. One way to do this is to install the miniconda distribution of Anaconda, which is a smaller version. We can then use our environment.yml file to install exactly what we will need.
```
Comment (tmthyjames) [inline AWS/ec2_practice/README.md]: typo - should be Once instead of One

**Review by tmthyjames — COMMENTED** [review]
Great stuff!

**Review by raom1 — APPROVED** [review]
LGTM! One question, I read somewhere that once you start using `sudo` you have to keep using it. Is there any down side to using `sudo`? This is mainly for my own curiosity, as I have avoided using `sudo` as much as possible up to this point.

File: AWS/ec2_practice/README.md (hunk 1 above)
Comment (mcvholloway) [reply AWS/ec2_practice/README.md]: Fixed!

**Comment by mcvholloway** [comment]
@raom1 You do need the root user permissions that `sudo` grants when using `apt`. I don't know of any downside to using `sudo` as long as you know what you're doing and don't use it to run a command that corrupts your OS or something.

**Pull request merged**